client/src/common/types/schema.ts

```diff
@@ -0,0 +1,58 @@
+type Category = number | string | boolean;
+
+export interface AnnotationColumnSchema {
+  categories?: Category[];
+  name: string;
+  type: "string" | "float32" | "int32" | "categorical" | "boolean";
+  writable: boolean;
+}
+
+interface DataFrameSchema {
```

Comment:
This is more properly called something like `XMatrixSchema`

the `dataframe` key in the JSON was an unfortunate (legacy) misunderstanding by the author (me :-) that we never fixed!!!